Core/Main/itkTransformixFilter.hxx

```diff
@@ -190,6 +196,77 @@ TransformixFilter<TMovingImage>::GenerateData()
     }
   }
 
+  if (m_Transform)
+  {
+    // Adjust the local transformParameterMap according to this m_Transform.
+
+    const auto transformToMap = [](const itk::TransformBase & transform, auto & transformParameterMap) {
+      const auto convertToParameterValues = [](const itk::OptimizerParameters<double> & optimizerParameters) {
+        ParameterValueVectorType parameterValues(optimizerParameters.size());
+        std::transform(optimizerParameters.begin(),
+                       optimizerParameters.end(),
+                       parameterValues.begin(),
+                       itk::NumberToString<double>{});
+        return parameterValues;
+      };
+
+      transformParameterMap["ITKTransformFixedParameters"] = convertToParameterValues(transform.GetFixedParameters());
+      transformParameterMap["ITKTransformParameters"] = convertToParameterValues(transform.GetParameters());
+      transformParameterMap["ITKTransformType"] = { transform.GetTransformTypeAsString() };
+      transformParameterMap["Transform"] = { elx::TransformIO::ConvertITKNameOfClassToElastixClassName(
+        transform.GetNameOfClass()) };
+    };
+    const auto compositeTransform =
+      dynamic_cast<const CompositeTransform<double, MovingImageDimension> *>(&*m_Transform);
+
+    if (compositeTransform)
+    {
+      const auto & transformQueue = compositeTransform->GetTransformQueue();
+
+      const auto numberOfTransforms = transformQueue.size();
+
+      if (numberOfTransforms == 0)
+      {
+        itkExceptionMacro(
+          "The specified composite transform has no subtransforms! At least one subtransform is required!");
+      }
+
+      if (numberOfTransforms != numberOfTransformParameterMaps)
+      {
+        if (numberOfTransformParameterMaps == 0)
+        {
+          transformParameterMapVector.resize(numberOfTransforms);
+        }
+        else
+        {
+          // The last TransformParameterMap is special, as it needs to be used for the final transformation.
+          const auto lastTransformParameterMap = transformParameterMapVector.back();
+          transformParameterMapVector.resize(numberOfTransforms);
+          transformParameterMapVector.back() = lastTransformParameterMap;
+        }
+      }
```

Comment:
@mstaring @stefanklein Please check if you like this logic:
 - The user-specified compositeTransform mus have _at least_ one subtransform (otherwise throw an exception)
 - If the number of parameter maps is zero, resize the vecor of maps to the number of subtransforms.
 - Otherwise, if the number of subtransforms of the compositeTransform is different from the number of parameter maps, add maps to the end, or remove maps from the end, to make these numbers the same, and then, copy the original "last parameter map" to the last entry of the vector of maps (the "back").

Is that the intended behavior?

Reply:
Discussed with Marius (@mstaring) at the office: If the number of parameter maps is zero, an exception should be thrown. For the other cases, the behavior was described as intended.